File: packages/scanner/tests/fixtures/VCS-012/vulnerable.js
const crypto = require("crypto");

// MD5 — broken
function hashPassword(pw) {
  return crypto.createHash("md5").update(pw).digest("hex");
}

// SHA-1 — deprecated
function fingerprint(data) {
  return crypto.createHash("sha1").update(data).digest("hex");
}

// DES cipher — broken
const cipher = crypto.createCipheriv("des-cbc", key, iv);

// RC4 — broken
const rc4 = crypto.createCipheriv("rc4", key, "");

// AES-ECB — pattern leakage
const ecb = crypto.createCipheriv("aes-256-ecb", key, "");

File: packages/scanner/tests/fixtures/VCS-012/vulnerable.py
import hashlib
from Crypto.Cipher import AES, DES

# MD5 — broken
def hash_password(pw):
    return hashlib.md5(pw.encode()).hexdigest()

# SHA1 — deprecated
def fingerprint(data):
    return hashlib.sha1(data).hexdigest()

# DES — 56-bit key, broken
cipher = DES.new(key, DES.MODE_CBC)

# AES-ECB — does not hide patterns
ecb = AES.new(key, AES.MODE_ECB)
